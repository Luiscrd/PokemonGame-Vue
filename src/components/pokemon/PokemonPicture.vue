<template>
    <div class="pokemon-container mt-3">
        <img :src="imgSrc" class="hidden-pokemon" alt="pokemon">
        <img v-if="showPokemon" :src="imgSrc" class="fade-in" alt="pokemon">
    </div>
    

</template>

<script>

export default {

    props:{

        pokemonId: {
            type: Number,
            required: true

        },

        showPokemon: {
            type: Boolean,
            required: true,
            default: false

        }
    },

    computed: {
        imgSrc() {
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${ this.pokemonId }.svg`
        }
    }

}
</script>

<style scoped>
.pokemon-container {
    height: 250px;
    position: relative;
}
img {
  height: 250px;
  position: absolute;
  width: 100%;
  margin-left: -50%;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}
</style>
