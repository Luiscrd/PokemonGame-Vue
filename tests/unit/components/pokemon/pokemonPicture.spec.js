import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/pokemon/PokemonPicture.vue'

describe('PokemonOptions Component', () => {

    let wrapper;

    test('Debe de hacer mach con el snapshot', () => {

        wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false
            }
        })
    
        expect(wrapper.html()).toMatchSnapshot()

    })

    test('Debe de mostrar la imagen oculta y el Pokémon 100', () => {

        wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })

        const img = wrapper.find('img')

        expect(img.exists()).toBeTruthy()

        expect(img.classes('hidden-pokemon')).toBeTruthy()

        expect(img.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')

    })

    test('Debe de mostrar el Pokémon si showPokemon: true', () => {

        wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })

        const [img1, img2] = wrapper.findAll('img')

        expect(img1.exists()).toBeTruthy()

        expect(img2.exists()).toBeTruthy()

        expect(img2.classes('fade-in')).toBeTruthy()

        expect(img2.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')
    
    })

})