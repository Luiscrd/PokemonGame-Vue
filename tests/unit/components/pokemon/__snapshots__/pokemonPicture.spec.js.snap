// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component Debe de hacer mach con el snapshot 1`] = `
<div class="pokemon-container mt-3"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="pokemon">
  <!--v-if-->
</div>
`;
